### PR TITLE
Feature/6.3/tdi 37456

### PR DESCRIPTION
**Pull request by wwang-talend**

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/445/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/445/)

**Comment by wwang-talend** [comment]
the test fail as some env issue for jira component, so wait for it.

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/452/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/452/)

File: components-jdbc/src/main/java/org/talend/components/jdbc/tjdbcrow/TJDBCRowDefinition.java [hunk 1]
```diff
@@ -41,19 +45,40 @@ public TJDBCRowDefinition() {
...
     // TODO add more return properties
+    @SuppressWarnings("rawtypes")
     @Override
     public Property[] getReturnProperties() {
         return new Property[] { RETURN_ERROR_MESSAGE_PROP, RETURN_TOTAL_RECORD_COUNT_PROP };
     }
 
     @Override
     public RuntimeInfo getRuntimeInfo(Properties properties, ConnectorTopology connectorTopology) {
+        if (connectorTopology == ConnectorTopology.OUTGOING) {
```
Comment (sgandon) [inline components-jdbc/src/main/java/org/talend/components/jdbc/tjdbcrow/TJDBCRowDefinition.java]: please use a Switch/case statement here.

File: components-jdbc/src/main/java/org/talend/components/jdbc/runtime/JDBCTemplate.java [hunk 2]
```diff
@@ -45,4 +65,126 @@ public static Connection createConnection(JDBCConnectionModule properties) throw
         return result;
     }
 
+    public static RuntimeInfo createCommonRuntime(ClassLoader classLoader, final Properties properties,
+            final String runtimeClassName) {
+        DependenciesReader dependenciesReader = new DependenciesReader("org.talend.components", "components-jdbc");
+        final Set<String> default_dependencies;
+        try {
+            default_dependencies = dependenciesReader.getDependencies(classLoader);
```
Comment (sgandon) [inline components-jdbc/src/main/java/org/talend/components/jdbc/runtime/JDBCTemplate.java]: I don't understand why you get the dependencies here and not below ** getMavenUrlDependencies** method ?

File: components-jdbc/src/main/java/org/talend/components/jdbc/tjdbcrow/TJDBCRowDefinition.java (hunk 1 above)
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/tjdbcrow/TJDBCRowDefinition.java]: done

File: components-jdbc/src/main/java/org/talend/components/jdbc/runtime/JDBCTemplate.java (hunk 2 above)
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/runtime/JDBCTemplate.java]: you are right, will move it into getMavenUrlDependencies method

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/458/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/458/)

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/461/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/461/)

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/472/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/472/)

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/476/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/476/)

**Pull request closed without merging**